### PR TITLE
5759 revoke grant on cdb conf to cdb organization member group role member name

### LINKED ISSUE #5759: Revoke grant on cdb_conf to CDB_Organization_Member_Group_Role_Member_Name
During organization creation CDB_Organization_Member_Group_Role_Member_Name role is granted select on cartodb.*, which conflicts with cdb_conf, since only superadmin should have access.

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #5759 and #5768. @Kartones @rafatower CR this, please.

### CODE REVIEW

File: app/models/organization.rb [hunk 1]
```diff
@@ -248,6 +248,41 @@ def database_name
     owner ? owner.database_name : nil
   end
 
+  def revoke_cdb_conf_access
+    return unless owner
+
+    errors = []
+
+    roles = users.collect(&:database_username)
```
Comment (rafatower) [inline app/models/organization.rb]: for once I agree with hound :D

**Comment by rafatower** [comment]
where is the grant done? there's a point of failure if we need to revoke explicitly

File: app/models/user.rb [hunk 2]
```diff
@@ -2260,11 +2264,22 @@ def grant_all_on_database_queries
...
+    queries.concat(revoke_permissions_on_cartodb_conf_queries(granted_user)) if schema == 'cartodb'
+
+    queries
+  end
+
+  def revoke_permissions_on_cartodb_conf_queries(db_user)
+    # TODO: remove the check after extension install (#4924 merge)
+    return [] if Rails.env.test?
+
+    [ "REVOKE ALL ON TABLE cartodb.CDB_CONF FROM \"#{db_user}\"" ]
```
Comment (rafatower) [inline app/models/user.rb]: I would move this to the extension, as part of the installation. Otherwise there's a security flaw.

**Comment by rafatower** [comment]
> An extension's SQL script files can contain any SQL commands, except for transaction control commands (BEGIN, COMMIT, etc) and commands that cannot be executed inside a transaction block (such as VACUUM). This is because the script files are implicitly executed within a transaction block.

so the safest move is to have the creation and grant/revoke of permissions within the extension, so it is guaranteed there's no opportunity window for accessing the table in between.

**Comment by juanignaciosl** [comment]
@rafatower : 

1.- the grant is done at `set_user_privileges_in_cartodb_schema`, which uses `grant_read_on_schema_queries` passing `'cartodb'` schema. If I revoked it at the grant place other function could call the second with `'cartodb'` and it would screw it again, that's why I moved it there.

2.- it must be at User, because new user creation might grant it :(

File: app/models/user.rb [hunk 3]
```diff
@@ -2260,11 +2264,22 @@ def grant_all_on_database_queries
...
   def grant_read_on_schema_queries(schema, db_user = nil)
     granted_user = db_user.nil? ? self.database_username : db_user
-    [
+
+    queries = [
       "GRANT USAGE ON SCHEMA \"#{schema}\" TO \"#{granted_user}\"",
       "GRANT EXECUTE ON ALL FUNCTIONS IN SCHEMA \"#{schema}\" TO \"#{granted_user}\"",
       "GRANT SELECT ON ALL TABLES IN SCHEMA \"#{schema}\" TO \"#{granted_user}\""
     ]
+    queries.concat(revoke_permissions_on_cartodb_conf_queries(granted_user)) if schema == 'cartodb'
```
Comment (rafatower) [inline app/models/user.rb]: the gist of this changeset

**Comment by rafatower** [comment]
I see. :+1: 

bonus point: I would make the method `grant_read_on_schema_queries` to private section as it is inherently dangerous and only used from `User`.

**Comment by Kartones** [comment]
Arriving late to the party, so nothing to add to Rafa's comments about security...
 
I still dislike to see SQL on Ruby but considering the painful extension release process not much can be done, so... :+1:

**Comment by juanignaciosl** [comment]
I'd need a second CR, @Kartones , since I added the needed fix for CartoDB/cartodb#5759 . I didn't realize because editor uses table owner connection for editions, so it seemed to be done in my tests :(

**Comment by Kartones** [comment]
Checked the diffs since last CR and :+1: , is that enough or want me to fully check everything?

**Comment by juanignaciosl** [comment]
Great, thanks. I'll prepare the main branch and ping you in case you want to take a last look. That's a huge PR but maybe some quick scaning worths it. Thanks!

**Pull request merged**